src/com/projectkorra/projectkorra/event/AbilityDamageEntityEvent.java

```diff
@@ -27,11 +28,13 @@ public class AbilityDamageEntityEvent extends Event implements Cancellable {
 	/**
 	 * Create a new AbilityDamageEntityEvent
 	 *
+	 * @param source The player who used the damaging move
 	 * @param entity The entity that was damaged
 	 * @param ability The damaging ability
 	 * @param damage The amount of damage done
 	 */
-	public AbilityDamageEntityEvent(final Entity entity, final Ability ability, final double damage, final boolean ignoreArmor) {
+	public AbilityDamageEntityEvent(final Player source, final Entity entity, final Ability ability, final double damage, final boolean ignoreArmor) {
```

Comment:
Okay, these changes I disagree with
1. This will break any plugin currently using the event
2. The player source can already be gotten from `ability.getPlayer()`

Reply:
I don't think it will break anything? Maybe im missing something. 2. The problem is that the ability is fake in regards to the FireDamageTimer... While the constructor is now deprecated, we cant ensure everyone will switch over (even though I did for the Fire Classes... I know a some plugins like Jedcore also use the old constructor), so, until we can be sure that it won't break on large scale for everyone, I think we need to use the source given, because we can't trust the ability.

Reply:
Correct, we can't ensure everyone will move over. But ones like JedCore can easily be PRd towards to fix the bug.

Any plugin that creates the event will break, and even if they just use the event, they will still be using the wrong player. All up, this does not completely fix the bug for anyone without them updating _something_. Therefore, I think it's better to leave as is and just have them use the correct timer constructor